File: app/src/main/java/com/bebediary/memo/memodb/dao/NoteDao.java
package com.bebediary.memo.memodb.dao;

import android.arch.persistence.room.*;
import com.bebediary.memo.memodb.model.Note;
import com.bebediary.util.Constants;

import java.util.List;


@Dao
public interface NoteDao {

    @Query("SELECT * FROM "+ Constants.TABLE_NAME_NOTE)
    List<Note> getNotes();

    /*
     * Insert the object in database
     * @param note, object to be inserted
     */
    @Insert
    long insertNote(Note note);

    /*
     * update the object in database
     * @param note, object to be updated
     */
    @Update
    void updateNote(Note repos);

    /*
     * delete the object from database
     * @param note, object to be deleted
     */
    @Delete
    void deleteNote(Note note);

    // Note... is varargs, here note is an array
    /*
     * delete list of objects from database
     * @param note, array of oject to be deleted
     */
    @Delete
    void deleteNotes(Note... note);

}

File: app/src/main/java/com/bebediary/database/dao/NoteDao.kt
package com.bebediary.database.dao

import androidx.room.Dao
import androidx.room.Query
import com.bebediary.database.entity.Note
import io.reactivex.Flowable
import io.reactivex.Observable
import io.reactivex.Single

@Dao
interface NoteDao : BaseDao<Note> {

    @Query("SELECT * FROM note WHERE note.babyId = :babyId")
    fun getNotes(babyId: Long): Flowable<List<Note>>

    @Query("SELECT * FROM note WHERE note.id = :id")
    fun getNoteById(id: Long): Single<Note>
}